iml-gui/crate/src/components/command_modal.rs

```diff
@@ -193,6 +197,23 @@ pub fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg, GMsg>)
                 schedule_fetch_tree(model, orders);
             }
         }
+        Msg::CancelJob(job_id) => {
+            if let Some(job) = model.jobs.get(&job_id) {
+                if let Some(ct) = find_cancel_transition(job) {
+                    if model.cancelling_jobs.insert(job_id) {
+                        let fut = apply_job_transition(job_id, ct.clone());
```

Comment:
nit, instead of passing + cloning the transition, you could just pass the state field instead.

Reply:
After some discussion the small extra space was neglected to keep the signature more typed.